alb.tf

```diff
@@ -1,154 +1,254 @@
+/**
+* https://github.com/terraform-aws-modules/terraform-aws-alb
+*/
+
 data "aws_lb" "public" {
-  count = var.alb_attach_public_target_group ? 1 : 0
-  name  = "public"
+  name = "public"
 }
 
-data "aws_lb_listener" "public" {
-  count             = var.alb_attach_public_target_group ? 1 : 0
-  load_balancer_arn = data.aws_lb.public[count.index].arn
-  port              = 443
-}
+resource "aws_alb_target_group" "main" {
+  count = length(var.target_groups)
 
-data "aws_lb_listener" "public_80" {
-  count             = var.alb_attach_public_target_group ? 1 : 0
-  load_balancer_arn = data.aws_lb.public[count.index].arn
-  port              = 80
-}
+  name        = lookup(var.target_groups[count.index], "name", null)
+  name_prefix = lookup(var.target_groups[count.index], "name_prefix", null)
 
-resource "aws_alb_target_group" "public" {
-  count = var.alb_attach_public_target_group ? 1 : 0
+  vpc_id           = data.aws_vpc.selected.id
```

Comment:
does it makes sense to refactor this to a required variable as well instead of hardcoding this to `main`?

Reply:
👍